### DISCUSSION TITLE
How to reduce PNG image size or compress the PNG image by using ImageMagick in PHP ?

### DISCUSSION THREAD

**Discussion by ravitejamsr** [ImageMagick/ImageMagick#2277] [Help]
Hi All,

I was trying to compress the PNG image but its reducing in 30 KB . Below is my code

```
 $i->setImageFormat("png");
 $i->setImageCompression(Imagick::COMPRESSION_ZIP);
 $i->setOption('png:compression-level',7);
 $i->setOption('png:compression-strategy',4);
 $i->setOption('png:compression-filter',4);
 $i->setOption('png:format','png8');
 $i->setOption('png:bit-depth',8);
 $i->stripImage();
```
Can any one help me out of this

**Comment by snibgo** [comment]
Is there a question? I don't understand what you want, or if something goes wrong, or whatever.

**Reply by ravitejamsr** [reply]
@snibgo  I was trying to compress the PNG format image with Imagick when i compress the 5.5 MB image size is reducing by 30KB like that and the result image is 5.2 MB but i want to compress the image by 3MB

**Comment by snibgo** [comment]
Different numbers for level, strategy and filter may improve the compression.

IM does not contain code to automatically explore all combinations. The program "pngcrush" can do that.
